src/askui/tools/android/agent_os.py

```diff
@@ -375,9 +375,9 @@ def set_device_by_index(self, device_index: int = 0) -> None:
         raise NotImplementedError
 
     @abstractmethod
-    def set_device_by_name(self, device_name: str) -> None:
+    def set_device_by_serial_number(self, device_serial_number: str) -> None:
```

Comment:
As it is already set in the method name (and `sn` is a common abbreviation), i would go for a very short parameter name (also in the method of the `AndroidVisionagent`):
```suggestion
def set_device_by_serial_number(self, dsn: str) -> None:
```

or 

```suggestion
def set_device_by_serial_number(self, device_sn: str) -> None:
```

or 

```suggestion
def set_device_by_serial_number(self, value: str) -> None:
```